### PR TITLE
Adding support for indexed normals on import 

### LINKED ISSUE #826: The normals dont get imported properly when they are indexed
When importing a model with authored normals, but the normals are stored indexed they dont look correct in maya

Store the normals as an indexed primvar:
  normal3f[] primvars:normals = [(-0.00343, 0.99999, 0), ...
                interpolation = "faceVarying"
            )
    int[] primvars:normals:indices = [2, ...
          

Load this into maya pr  119,  the normals will look off.

Thanks,
Koen

image

### PR DESCRIPTION
**Pull request by koen-v**
The original code only looked for  normals using the GetNormalsAttr call. According to the specs, the primvars:normals primvar takes precedence ( and this is what we use to allow for indexed normals.)

http://graphics.pixar.com/usd/docs/api/class_usd_geom_point_based.html#ac5427209fe38570a75036a3cb7ebb553

This modified code will look for the primvar normals first and if not found will use the previous behavior.

Fixes #826

### CODE REVIEW

**Comment by kxl-adsk** [comment]
Hi @koen-v. Thank you for opening your first pull-request and contribution to the maya-usd project. I don't believe we have CLA signed from you - please have a look at https://github.com/Autodesk/maya-usd/blob/dev/doc/CONTRIBUTING.md

You may also notice that clang-format-lint check failed. We use clang-format (version 10.0.0) to keep the formatting consistent across the repo. Make sure to apply clang-format.

Finally, it would be great to have a test added for your latest change. This will guarantee there are no regressions in what you are fixing. I will let reviewers make suggestions here.

File: lib/mayaUsd/fileio/translators/translatorMesh.cpp [hunk 1]
```diff
@@ -127,7 +129,17 @@ TranslatorMeshRead::TranslatorMeshRead(
     }
 
     mesh.GetPointsAttr().Get(&points, pointsTimeSample);
-    mesh.GetNormalsAttr().Get(&normals, normalsTimeSample);
+
+    /* If 'normals' and 'primvars:normals' are both specified, the latter has precedence. */
+    UsdGeomPrimvar primvar = mesh.GetPrimvar(TfToken("normals"));
```
Comment (JGamache-autodesk) [inline lib/mayaUsd/fileio/translators/translatorMesh.cpp:134]: Please use `UsdGeomTokens->normals`. Everything else looks fine. Thanks!

**Review by JGamache-autodesk — APPROVED** [review]

**Comment by JGamache-autodesk** [comment]
For testing, you can add a normal stream to the test mesh found in `test\lib\usd\translators\UsdImportMeshTest\Mesh.usda` and check that they are correctly imported in `test\lib\usd\translators\testUsdImportMesh.py`.

**Comment by koen-v** [comment]
Thanks for looking at this. I am running the contributor agreement by our legal department at the moment. I will look at the clang formatting and the comments regarding the tokens and testing. 

For the testing I will add a new mesh to the usda as to  not shadow testing the regular normals.

Cheers,
Koen

**Comment by kxl-adsk** [comment]
@koen-v Can you give us an update on where you are with the CLA process? Feel free to contact me directly if you have questions or would prefer to discuss this offline.

**Comment by koen-v** [comment]
Thanks for checking in.

It has been approved and mostly made it's way through legal I believe, let me check.

Cheers,
Koen

**Comment by koen-v** [comment]
Sorry something went wrong with my branch seems like.

I will remove this request

**Comment by kxl-adsk** [comment]
No worries. Are you going to open it again?

**Comment by koen-v** [comment]
Yes I will, I am kind of new to git and all these later commits that where not suited for integration all of a sudden popped up in the pull request. I guess the easiest will be to create a fork just for these changes that I can close once integrated. I'll keep you posted.

Cheers,
koen

**Pull request closed without merging**